docs/gen-ai/gen-ai-events.md

```diff
@@ -0,0 +1,379 @@
+|---|---|---|---|---|
+| `content` | string | The contents of the system message. | `"You're a friendly bot that answers questions about OpenTelemetry."` | `Opt-In` |
+
+## User message
+
+This event describes the prompt message specified by the user.
+
+The event name MUST be `gen_ai.user.message`.
+
+| Body Field | Type | Description | Examples | Requirement Level |
+|---|---|---|---|---|
+| `content` | `AnyValue` | The contents of the user message. | `What telemetry is reported by OpenAI instrumentations?` | `Opt-In` |
+
+## Assistant message
+
+This event describes the assistant message.
+
+The event name MUST be `gen_ai.assistant.message`.
+
+| Body Field   | Type                           | Description                            | Examples                                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|--------------|--------------------------------|----------------------------------------|-------------------------------------------------|-------------------|
+| `content`    | `AnyValue`                     | The contents of the assistant message. | `Spans, events, metrics defined by the GenAI semantic conventions.` | `Opt-In` |
+| `tool_calls` | [ToolCall](#toolcall-object)[] | The tool calls generated by the model, such as function calls. | `[{"id":"call_mszuSIzqtI65i1wAUOE8w5H4", "function":{"name":"get_link_to_otel_semconv", "arguments":"{\"semconv\":\"gen_ai\"}"}, "type":"function"}]` | `Conditionally Required`: if available |
+
+### `ToolCall` object
+
+| Body Field | Type                           | Description                            | Examples                                        | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|------------|--------------------------------|----------------------------------------|-------------------------------------------------|-------------------|
+| `id`       | string                         | The id of the tool call                | `call_mszuSIzqtI65i1wAUOE8w5H4`                 | `Required`        |
+| `type`     | string                         | The type of the tool                   | `function`                                      | `Required`        |
+
+### `Function` object
+
+| Body Field  | Type       | Description                            | Examples                   | [Requirement Level](https://opentelemetry.io/docs/specs/semconv/general/attribute-requirement-level/) |
+|-------------|------------|----------------------------------------|----------------------------|-------------------|
+| `name`      | string     | The name of the function to call       | `get_link_to_otel_semconv` | `Required`        |
+| `arguments` | `AnyValue` | The arguments to pass the the function | `{"semconv": "gen_ai"}`    | `Opt-In`          |
+
+## Tool message
+
+This event describes the output of the tool or function submitted back to the model.
```

Comment:
> While there is an example, fields or events themselves have no description of what a tool call is or any implications of it. In anthropic "All tools are user-provided", so there are no server-side tools. This means it is a very good question, if these should be spans or not.. more on this later.

This PR describes how tool calls appear in prompts and completions. If server-side functions need extra fields, we can add them incrementally.

This PR does not describe (and does not intend to) describe how applications should record tool execution.

If you believe it's not clear or need additional explanations, the fastest way to achieve it would be to leave a specific suggestions at specific lines of this PR.